From 7e3d116eeb11da4c75adc83c90be7cde2c05d37b Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Fri, 25 Sep 2026 17:26:21 +0200
Subject: [PATCH 1/1] loader/efi/linux: print image start and size

When starting an EFI payload grub_arch_efi_linux_boot_image() currently
prints the handle of the loaded image. This information is insufficient
to debug crashes.

Add the image base and the image size to the debug output.

Here is an example output:

  grub_arch_efi_linux_boot_image:293:linux: starting image 0xbef01390
  grub_arch_efi_linux_boot_image:294:linux: image base 0xbbdc6000
  grub_arch_efi_linux_boot_image:295:linux: image size 0x3200

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
---
 grub-core/loader/efi/linux.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/grub-core/loader/efi/linux.c b/grub-core/loader/efi/linux.c
index d9118b21f..f5d413968 100644
--- a/grub-core/loader/efi/linux.c
+++ b/grub-core/loader/efi/linux.c
@@ -291,6 +291,9 @@ grub_arch_efi_linux_boot_image (grub_addr_t addr, grub_size_t size, char *args)
   loaded_image->load_options_size = len * sizeof (grub_efi_char16_t);
 
   grub_dprintf ("linux", "starting image %p\n", image_handle);
+  grub_dprintf ("linux", "image base %p\n", loaded_image->image_base);
+  grub_dprintf ("linux", "image size 0x%llx\n",
+		(unsigned long long) loaded_image->image_size);
   status = grub_efi_start_image (image_handle, 0, NULL);
 
   /* When successful, not reached */
-- 
2.53.0

